File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mx=0;
        int mn=prices[0];
        int p=0;
        int n=prices.size();
        for(int i=1;i<n;i++)
        {
            mx=max(mx,prices[i]-mn);
            mn=min(prices[i],mn);
        }
        return mx;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let n=prices.length;;
    let mx=0;
    let mn=prices[0];
    for(let i=1;i<n;i++)
        {
            mx=Math.max(mx,prices[i]-mn);
            mn=Math.min(mn,prices[i]);
        }
    return mx;
};